include/seastar/core/semaphore.hh

```diff
@@ -249,6 +249,26 @@ public:
     {
         static_assert(std::is_nothrow_move_constructible_v<expiry_handler>);
     }
+    basic_semaphore(basic_semaphore&& x) noexcept(std::is_nothrow_move_constructible_v<exception_factory>)
+        : exception_factory(std::move(x))
+        , _count(x._count)
+        , _wait_list(expiry_handler{*this})
+    {
```

Comment:
Should we move/assign _ex too? moves should happen before _ex is assigned anything, but better to be safe.

Reply:
yeah, makes sense.

Reply:
done in next version.
plus, moved also the used flag.